diff --git a/build/modules/dif-presentation-exchange/DifPresentationExchangeService.js b/build/modules/dif-presentation-exchange/DifPresentationExchangeService.js
index 12df3a49c3e7b1774fa18ef1a4e45ebb600d2da4..ba41b17660c2a4a5a53a1428c27bc304d662ed95 100644
--- a/build/modules/dif-presentation-exchange/DifPresentationExchangeService.js
+++ b/build/modules/dif-presentation-exchange/DifPresentationExchangeService.js
@@ -178,7 +178,18 @@ let DifPresentationExchangeService = class DifPresentationExchangeService {
             presentationSubmission.descriptor_map.push(...descriptorMap);
         });
         return {
-            verifiablePresentations: verifiablePresentationResultsWithFormat.flatMap((resultWithFormat) => resultWithFormat.verifiablePresentationResult.verifiablePresentations.map((vp) => (0, utils_2.getVerifiablePresentationFromEncoded)(agentContext, vp))),
+            verifiablePresentations: verifiablePresentationResultsWithFormat.flatMap((resultWithFormat) => {
+                if (resultWithFormat.verifiablePresentationResult.verifiablePresentations) {
+                    return resultWithFormat.verifiablePresentationResult.verifiablePresentations.map((vp) =>
+                        (0, utils_2.getVerifiablePresentationFromEncoded)(agentContext, vp)
+                    )
+                } else {
+                return (0, utils_2.getVerifiablePresentationFromEncoded)(
+                    agentContext,
+                    resultWithFormat.verifiablePresentationResult.verifiablePresentation
+                )
+                }
+            }),
             presentationSubmission,
             presentationSubmissionLocation: verifiablePresentationResultsWithFormat[0].verifiablePresentationResult.presentationSubmissionLocation,
         };
