diff --git a/build/openid4vc-holder/OpenId4VciHolderService.js b/build/openid4vc-holder/OpenId4VciHolderService.js
index 2d0149b8b4e2bd63ec7f52bd419ec6ac3db689ea..d81b30b51345d21e5ad5a253137b284381fc81e4 100644
--- a/build/openid4vc-holder/OpenId4VciHolderService.js
+++ b/build/openid4vc-holder/OpenId4VciHolderService.js
@@ -154,11 +154,11 @@ let OpenId4VciHolderService = class OpenId4VciHolderService {
     async sendNotification(options) {
         const { notificationMetadata, notificationEvent } = options;
         const { notificationId, notificationEndpoint } = notificationMetadata;
-        const response = await (0, oid4vci_common_1.post)(notificationEndpoint, { notification_id: notificationId, event: notificationEvent }, {
+        const response = await (0, oid4vci_common_1.post)(notificationEndpoint, JSON.stringify({ notification_id: notificationId, event: notificationEvent }), {
             bearerToken: options.accessToken,
             contentType: 'application/json',
         });
-        if (!response.successBody) {
+        if (response.error) {
             throw new core_1.CredoError(`Failed to send notification event '${notificationId}' to '${notificationEndpoint}'`);
         }
     }
