diff --git a/build/anoncreds/utils/transform.js b/build/anoncreds/utils/transform.js
index c38801e690b48598ae80df5466f7bffc7e587d04..57ada8e9486023caa04ab274916df52c9b73d85e 100644
--- a/build/anoncreds/utils/transform.js
+++ b/build/anoncreds/utils/transform.js
@@ -13,7 +13,7 @@ function anonCredsRevocationStatusListFromIndyVdr(revocationRegistryDefinitionId
     // Check whether the highest delta index is supported in the `maxCredNum` field of the
     // revocation registry definition. This will likely also be checked on other levels as well
     // by the ledger or the indy-vdr library itself
-    if (Math.max(...delta.issued, ...delta.revoked) >= revocationRegistryDefinition.value.maxCredNum) {
+    if (Math.max(...delta.issued, ...delta.revoked) > revocationRegistryDefinition.value.maxCredNum) {
         throw new core_1.CredoError(`Highest delta index '${Math.max(...delta.issued, ...delta.revoked)}' is too large for the Revocation registry maxCredNum '${revocationRegistryDefinition.value.maxCredNum}' `);
     }
     // 0 means unrevoked, 1 means revoked
diff --git a/build/pool/IndyVdrPool.js b/build/pool/IndyVdrPool.js
index eab13b315dede8a2a7f7f56caa850405cf5f2f47..2109fb8eeae12c490b8233403b4445a5f6e0e465 100644
--- a/build/pool/IndyVdrPool.js
+++ b/build/pool/IndyVdrPool.js
@@ -17,7 +17,7 @@ class IndyVdrPool {
     }
     connect() {
         if (this._pool) {
-            throw new error_1.IndyVdrError('Cannot connect to pool, already connected.');
+            return;
         }
         this._pool = new indy_vdr_shared_1.PoolCreate({
             parameters: {
