diff --git a/build/transport/WsOutboundTransport.js b/build/transport/WsOutboundTransport.js
index ab9df20f9fedc4def4f486b9431ac669e899e609..72404426d64caed8f0392484ff76ea8716a991c2 100644
--- a/build/transport/WsOutboundTransport.js
+++ b/build/transport/WsOutboundTransport.js
@@ -43,7 +43,16 @@ class WsOutboundTransport {
         this.transportTable.forEach((socket) => {
             socket.removeEventListener('message', this.handleMessageEvent);
             if (socket.readyState !== this.WebSocketClass.CLOSED) {
-                stillOpenSocketClosingPromises.push(new Promise((resolve) => socket.once('close', resolve)));
+                stillOpenSocketClosingPromises.push(
+                    new Promise((resolve) => {
+                      const closeHandler = () => {
+                        socket.removeEventListener('close', closeHandler)
+                        resolve()
+                      }
+          
+                      socket.addEventListener('close', closeHandler)
+                    })
+                  );
                 socket.close();
             }
         });
