diff --git a/build/pool/IndyVdrPool.js b/build/pool/IndyVdrPool.js
index eab13b315dede8a2a7f7f56caa850405cf5f2f47..2109fb8eeae12c490b8233403b4445a5f6e0e465 100644
--- a/build/pool/IndyVdrPool.js
+++ b/build/pool/IndyVdrPool.js
@@ -17,7 +17,7 @@ class IndyVdrPool {
     }
     connect() {
         if (this._pool) {
-            throw new error_1.IndyVdrError('Cannot connect to pool, already connected.');
+            return;
         }
         this._pool = new indy_vdr_shared_1.PoolCreate({
             parameters: {
