diff --git a/android/CMakeLists.txt b/android/CMakeLists.txt
index 74b6aff9c7cf062724475a58b9030f655a939e18..4e337641ecc3fb06a9089ecc3ba90fe6f32165eb 100644
--- a/android/CMakeLists.txt
+++ b/android/CMakeLists.txt
@@ -5,6 +5,6 @@ set (LIB_NAME "ariesaskar")
 file (GLOB LIBASKAR_DIR "${CMAKE_SOURCE_DIR}/../native/mobile/android/${ANDROID_ABI}")
 
 set(CMAKE_VERBOSE_MAKEFILE ON)
-set(CMAKE_CXX_STANDARD 14)
+set(CMAKE_CXX_STANDARD 17)
 
 set (BUILD_DIR ${CMAKE_SOURCE_DIR}/build)
@@ -12,6 +12,13 @@ set (BUILD_DIR ${CMAKE_SOURCE_DIR}/build)
 if(${REACT_NATIVE_VERSION} GREATER_EQUAL 71)
   find_package(fbjni REQUIRED CONFIG)
   find_package(ReactAndroid REQUIRED CONFIG)
+  if(NOT TARGET ReactAndroid::reactnativejni)
+    if(TARGET ReactAndroid::reactnative)
+      add_library(ReactAndroid::reactnativejni ALIAS ReactAndroid::reactnative)
+    else()
+      message(FATAL_ERROR "ReactAndroid::reactnativejni (or reactnative) target not found")
+    endif()
+  endif()
 else()
   set (RN_SO_DIR ${NODE_MODULES_DIR}/react-native/ReactAndroid/src/main/jni/first-party/react/jni)
 endif()
diff --git a/android/build.gradle b/android/build.gradle
index 59e6410b844e3b85c750bca454f1c27cfadf8714..7cbf0f5ea95b907b5dfd9168ed2d2b2dd07b6ef7 100644
--- a/android/build.gradle
+++ b/android/build.gradle
@@ -110,6 +110,7 @@ android {
     excludes = ["**/libc++_shared.so", 
                 "**/libfbjni.so", 
                 "**/libjsi.so", 
+                "**/libreactnative.so",
                 "**/libreactnativejni.so", 
                 "**/libjscexecutor.so", 
     ]
