diff --git a/node_modules/@aries-framework/anoncreds-rs/build/AnonCredsRsModule.js b/node_modules/@aries-framework/anoncreds-rs/build/AnonCredsRsModule.js
index 55d55b8..4f5d79e 100644
--- a/node_modules/@aries-framework/anoncreds-rs/build/AnonCredsRsModule.js
+++ b/node_modules/@aries-framework/anoncreds-rs/build/AnonCredsRsModule.js
@@ -5,18 +5,6 @@ const anoncreds_1 = require("@aries-framework/anoncreds");
 const services_1 = require("./services");
 class AnonCredsRsModule {
     register(dependencyManager) {
-        try {
-            // eslint-disable-next-line import/no-extraneous-dependencies
-            require('@hyperledger/anoncreds-nodejs');
-        }
-        catch (error) {
-            try {
-                require('@hyperledger/anoncreds-react-native');
-            }
-            catch (error) {
-                throw new Error('Could not load anoncreds bindings');
-            }
-        }
         // Register services
         dependencyManager.registerSingleton(anoncreds_1.AnonCredsHolderServiceSymbol, services_1.AnonCredsRsHolderService);
         dependencyManager.registerSingleton(anoncreds_1.AnonCredsIssuerServiceSymbol, services_1.AnonCredsRsIssuerService);