diff --git a/node_modules/@aries-framework/indy-vdr/build/IndyVdrModule.js b/node_modules/@aries-framework/indy-vdr/build/IndyVdrModule.js
index 8220422..57348ef 100644
--- a/node_modules/@aries-framework/indy-vdr/build/IndyVdrModule.js
+++ b/node_modules/@aries-framework/indy-vdr/build/IndyVdrModule.js
@@ -11,18 +11,6 @@ class IndyVdrModule {
         this.config = new IndyVdrModuleConfig_1.IndyVdrModuleConfig(config);
     }
     register(dependencyManager) {
-        try {
-            // eslint-disable-next-line import/no-extraneous-dependencies
-            require('@hyperledger/indy-vdr-nodejs');
-        }
-        catch (error) {
-            try {
-                require('@hyperledger/indy-vdr-react-native');
-            }
-            catch (error) {
-                throw new Error('Error registering bindings for Indy VDR');
-            }
-        }
         // Config
         dependencyManager.registerInstance(IndyVdrModuleConfig_1.IndyVdrModuleConfig, this.config);
         // Services